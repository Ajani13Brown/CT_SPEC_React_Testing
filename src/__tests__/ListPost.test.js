import React from 'react';
import axios from 'axios';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ListPosts from '../Components/ListPost';

jest.mock('axios')

describe('ListPost Intergration test', () => {
    test("Post list Componet Test", async () => {
        
        render(<ListPosts />)
        await waitFor(() => {
            expect(axios.get).toHaveBeenCalledWith('https://jsonplaceholder.typicode.com/posts')


        const { asFragment } = render(<ListPosts />)
        expect(asFragment()).toMatchSnapshot()

        })

    })
})