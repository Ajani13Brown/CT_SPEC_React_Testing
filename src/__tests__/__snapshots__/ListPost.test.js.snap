// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPost Intergration test Post list Componet Test 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="text-center"
    >
      Post List
    </h2>
  </div>
</DocumentFragment>
`;
